packages/javascript/bh-shared-ui/src/hooks/useSearch/useSearch.tsx

```diff
@@ -67,6 +67,21 @@ export const getKeywordAndTypeValues = (
     return { keyword: keyword, type: type };
 };
 
+const validateNodeType = (type: string): ActiveDirectoryNodeKind | AzureNodeKind | undefined => {
+    let result = undefined;
+    Object.values(ActiveDirectoryNodeKind).forEach((activeDirectoryType: string) => {
+        if (activeDirectoryType.localeCompare(type, undefined, { sensitivity: 'base' }) === 0)
```

Comment:
Thats a very cool way to compare two strings! Im curious, is there an added benefit of using `localeCompare` instead of `type.toLowerCase() === activeDirectoryType.toLowerCase()` in this case? Looks like usually `localeCompare` is used in a sorting function but it totally works here as well, I guess Im just curious if Im missing something else going on!

Reply:
also seeing that this change was from the migrating into `bh-shared-ui` - feel free to ignore as well